code/datums/components/stationloving.dm

```diff
@@ -0,0 +1,69 @@
+/datum/component/stationloving
+	var/inform_admins = FALSE
+
+/datum/component/stationloving/Initialize(inform_admins = FALSE)
+	if(!ismovableatom(parent))
+		return COMPONENT_INCOMPATIBLE
+	RegisterSignal(list(COMSIG_MOVABLE_Z_CHANGED), .proc/check_in_bounds)
+	src.inform_admins = inform_admins
+	check_in_bounds() // Just in case something is being created outside of station/centcom
+
+/datum/component/stationloving/proc/relocate()
+	var/targetturf = find_safe_turf()
+	if(!targetturf)
+		if(GLOB.blobstart.len > 0)
+			targetturf = get_turf(pick(GLOB.blobstart))
+		else
+			throw EXCEPTION("Unable to find a blobstart landmark")
+
+	var/atom/movable/AM = parent
+	if(ismob(AM.loc))
+		var/mob/M = AM.loc
+		M.transferItemToLoc(AM, targetturf, TRUE)	//nodrops disks when?
+	else if(AM.loc.SendSignal(COMSIG_CONTAINS_STORAGE))
+		AM.loc.SendSignal(COMSIG_TRY_STORAGE_TAKE, src, targetturf, TRUE)
+	else
+		AM.forceMove(targetturf)
+	// move the disc, so ghosts remain orbiting it even if it's "destroyed"
+	return targetturf
+
+/datum/component/stationloving/proc/check_in_bounds()
+	if(in_bounds())
+		return
+	else
+		var/turf/currentturf = get_turf(src)
+		to_chat(get(parent, /mob), "<span class='danger'>You can't help but feel that you just lost something back there...</span>")
```

Comment:
?

Reply:
`get(atom/loc, type)` returns a parent of the atom `loc` that is of type `type`, this is the message that informs you you had a stationloving thing in your inventory when you went to lavaland or across a Z transition or whatever